test/server_test.go

```diff
@@ -48,8 +49,42 @@ func TestBasicEndToEnd(t *testing.T) {
 	assert.Equal(t, bookName, singleBook.Name)
 }
 
+func TestNextPageHeader(t *testing.T) {
+	s, controller := setupServer()
+	controller.pageSize = 2
+
+	c := client.New(s.URL)
+
+	_, err := c.CreateBook(context.Background(), &models.Book{
+		ID: int64(1), Name: "Test",
+	})
+	require.NoError(t, err)
+
+	// Make a raw HTTP request (i.e. don't use the client) so we can check the headers
+	resp, err := http.Get(fmt.Sprintf("%s/v1/books", s.URL))
+	assert.NoError(t, err)
+	resp.Body.Close()
+	assert.Equal(t, "", resp.Header.Get("X-Next-Page-Path"))
+
+	controller.pageSize = 1
+	_, err = c.CreateBook(context.Background(), &models.Book{
+		ID: int64(2), Name: "Second",
+	})
+	require.NoError(t, err)
+
+	// Now that there is a second page, X-Next-Page-Path should be set
+	resp, err = http.Get(fmt.Sprintf("%s/v1/books", s.URL))
```

Comment:
worth having the test set a query parameter that's not related to paging to make sure the URL gets set up correctly?